js/events/core/events_engine.js

```diff
@@ -614,6 +614,18 @@ eventsEngine.hover = {
     }
 };
 
+eventsEngine.visibility = {
+    on: ($el, shown, hiding, { namespace }) => {
```

Comment:
Doesn't that seem a little confusing to you?

We have to remember the order of the event handlers. Also, don't forget that some parts of EventsEngine have only one handler, while others have several handlers.

Perhaps we can unify the structure of the parameters, for example: `.on(#target, handler, options)` where `handler: function|Object` 
So we can group handlers this way:
```suggestion
    on: ($el, { shown, hiding }, { namespace }) => {
```
And use it like follows:
```js
visibilityEvents.on($el,
  {
    shown: shownHandler,
    hiding: hidingHandler
  },
  { namespace }
);
```